x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -308,11 +308,44 @@ func (k Keeper) CanValidateChain(ctx sdk.Context, chainID string, providerAddr t
 			!k.IsDenylisted(ctx, chainID, providerAddr))
 }
 
+// FulfillsMinStake returns true if the validator `providerAddr` has enough stake to validate chain `chainID`
+// by checking its staked tokens against the minimum stake required to validate the chain.
+func (k Keeper) FulfillsMinStake(ctx sdk.Context, chainID string, providerAddr types.ProviderConsAddress) bool {
+	minStake, found := k.GetMinStake(ctx, chainID)
+	if !found {
+		return true
+	}
+
+	validator, err := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerAddr.Address)
+	if err != nil {
+		return false
+	}
+
+	// validator has enough stake to validate the chain
+	return validator.GetBondedTokens().GTE(math.NewIntFromUint64(minStake))
+}
+
 // ComputeNextValidators computes the validators for the upcoming epoch based on the currently `bondedValidators`.
-func (k Keeper) ComputeNextValidators(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator) []types.ConsumerValidator {
+func (k Keeper) ComputeNextValidators(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator) []types.ConsensusValidator {
+	// sort the bonded validators by number of staked tokens in descending order
```

Comment:
Aren't those already sorted?  The way I understand this, is that when we call `ComputeNextValidators`, we pass `bondedValidators` that stem from `GetLastBondedValidatorsUtil` that should have sorted those already.

Reply:
Agreed. I have it for defensive-programming purposes here, since it will be inexpensive if the input is already sorted and means this function has less (implicit) assumptions

Reply:
lmk if you see a reason to change it